wagtail/core/blocks/field_block.py

```diff
@@ -22,6 +22,44 @@ class FieldBlock(Block):
     def id_for_label(self, prefix):
         return self.field.widget.id_for_label(prefix)
 
+    def prepare_value(self, value, errors=None):
+        from wagtail.admin.rich_text import DraftailRichTextArea
+        from wagtail.admin.widgets import AdminDateInput, AdminDateTimeInput
+
+        value = self.value_for_form(self.field.prepare_value(value))
+        widget = self.field.widget
+        if isinstance(self, RichTextBlock) \
+                and isinstance(widget, DraftailRichTextArea):
+            value = widget.format_value(value)
+        if isinstance(widget, (AdminDateInput, AdminDateTimeInput)):
+            value = widget.format_value(value)
+        return value
+
+    def prepare_for_react(self, parent_block, value,
+                          type_name=None, errors=None):
+        data = super().prepare_for_react(parent_block, value,
+                                         type_name=type_name, errors=errors)
+        if errors:
+            data['html'] = self.render_form(
+                value, prefix=Block.FIELD_NAME_TEMPLATE, errors=errors)
+        return data
+
+    def get_definition(self):
+        definition = super(FieldBlock, self).get_definition()
+        definition['html'] = self.render_form(self.get_default(),
+                                              prefix=self.FIELD_NAME_TEMPLATE)
+        title_template = self.get_title_template()
+        if title_template is not None:
+            definition['titleTemplate'] = title_template
+        return definition
+
+    def get_title_template(self):
+        if isinstance(self, (CharBlock, TextBlock, FloatBlock,
+                             DecimalBlock, RegexBlock, URLBlock,
+                             DateBlock, TimeBlock, DateTimeBlock,
+                             EmailBlock, IntegerBlock)):
```

Comment:
A list like this will be the source of bug reports because it will always miss some class, especially user created ones.

Reply:
Mmm… I don’t really see why, this method can be defined in any child class.
An alternative to this tuple would be to specify the same method in all of these blocks, but I’m not sure that’s really better.

Reply:
How about having a `use_as_title` flag defined on `FieldBlock`, defaulting to False but overrideable on subclasses, and checking that flag here?